update-cafe/src/main/kotlin/us/wedemy/eggeum/android/updatecafe/viewmodel/ProposeCafeInfoViewModel.kt

```diff
@@ -18,55 +19,85 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharedFlow
 import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import timber.log.Timber
 import us.wedemy.eggeum.android.domain.model.place.PlaceEntity
 import us.wedemy.eggeum.android.domain.model.place.ProductEntity
 import us.wedemy.eggeum.android.domain.usecase.GetPlaceUseCase
 import us.wedemy.eggeum.android.domain.usecase.UpsertPlaceUseCase
+import us.wedemy.eggeum.android.main.model.CafeDetailModel
+import us.wedemy.eggeum.android.updatecafe.ui.item.CafeInfoItem
 import us.wedemy.eggeum.android.updatecafe.ui.item.CafeMenuItem
 
 @HiltViewModel
 class ProposeCafeInfoViewModel @Inject constructor(
   private val getPlaceUseCase: GetPlaceUseCase,
   private val upsertPlaceUseCase: UpsertPlaceUseCase,
+  savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
+  private val selectedCafeInfo: CafeDetailModel = savedStateHandle[CAFE_DETAIL_INFO]
+    ?: throw IllegalArgumentException("CAFE_DETAIL_INFO is missing from SavedStateHandle")
```

Comment:
require 문을 사용하는게 더 kotlin 스러운 코드라 아래 처럼 바꾸는게 좋을 것 같습니다

```suggestion
private val selectedCafeInfo: CafeDetailModel = requireNotNull(savedStateHandle[CAFE_DETAIL_INFO]) { "cafeDetailInfo is required" }
```

Reply:
꿀팁 감사합니다.